### PR TITLE
Json schema v2

### PR DESCRIPTION
**Pull request by pieterbos**
JSON Schema improvements, as preparation to open a pull request on github.com/openEHR/specifications-ITS-JSON.

Done so far:
- Fixed a P_BMM -> BMM bug where package names were incorrect
- Added  a couple of types that can be put at the root of a json document. More discussion is needed here.
- Removed all generics parameters from type names, as concluded in https://discourse.openehr.org/t/correct-use-of-generic-types-in-xml-and-json/1504 (sorry, that discussion is SEC members only)
- Enabled the JSON Schema to be split in packages, with an easily configured naming strategy, for easier human readable results. Fully configurable, either with two defaults or a separate method.
- added a separate gradle directory to generate the json schema files easily.
- added $id parts

I opened a pull request with these adaptations at https://github.com/openEHR/specifications-ITS-JSON/pull/9

### CODE REVIEW

**Comment by pieterbos** [comment]
Pull request in standard has been merged. So, this should porbably go into Archie as well! @J3173 there's one relatively complex part, that is the split JSON files do not work witht he validator included before. I added a validator2. That's not a solution I like.

Possbile solutions to that:
1. Get rid of validator 2, and work with single file schema only
2. Switch fully to validator 2, making this a breaking change.

opinions?

File: tools/src/main/java/com/nedap/archie/json/JSONSchemaCreator.java [hunk 1]
```diff
@@ -360,12 +392,123 @@ private JsonObjectBuilder createType(String jsPrimitive) {
...
-        return jsonFactory.createObjectBuilder().add("$ref", "#/definitions/" + rootType);
+    private JsonObjectBuilder createRootlevelReference(JsonSchemaUri packageFileName, String type) {
+        BmmClass bmmClass = bmmModel.getClassDefinitions().get(type);
+        if(bmmClass == null) {
+            //TODO: a warning!
+            return jsonFactory.createObjectBuilder().add("$ref", "#/definitions/" + type);
+        } else {
+            JsonSchemaUri typeFileName = uriProvider.provideJsonSchemaUrl(bmmClass);
+            if(typeFileName == null) {
+                throw new RuntimeException();
```
Comment (J3173) [inline tools/src/main/java/com/nedap/archie/json/JSONSchemaCreator.java:403]: Can you add a short error message here?

File: tools/build.gradle [hunk 2]
```diff
@@ -16,4 +16,5 @@ dependencies {
 
   compile 'org.leadpony.justify:justify:3.1.0'
   compile 'org.glassfish:jakarta.json:2.0.1:module'
-}
\ No newline at end of file
+  implementation 'com.github.everit-org.json-schema:org.everit.json.schema:1.14.0'
```
Comment (J3173) [inline tools/build.gradle:19]:
```suggestion
  implementation 'com.github.erosb:everit-json-schema:1.14.0'
```
This variant seems to be the one available in Maven Central. This removes the need for configuring an extra repository.

Comment (pieterbos) [reply tools/build.gradle:19]: The jitpack one is the official location, plut it's not a new repository, it's already included in https://github.com/openEHR/archie/blob/master/build.gradle#L42

However, this json-schema part is not technically part of the archie module build, so is not a submodule, but 'just' a gradle module using archie. That is why it is included here.

Makes me think maybe I should try to make it an Archie submodule again, which would remove the need for this kind of separate config. Or a separate repo just for this script?

**Comment by pieterbos** [comment]
processed all review comments.

**Review by J3173 — APPROVED** [review]

**Pull request merged**